packages/mobile/src/components/user-link/UserLink.tsx

```diff
@@ -0,0 +1,36 @@
+import type { ID } from '@audius/common/models'
+import { cacheUsersSelectors } from '@audius/common/store'
+import { useSelector } from 'react-redux'
+
+import type { TextLinkProps } from '@audius/harmony-native'
+import { TextLink, Flex } from '@audius/harmony-native'
+import type { AppTabScreenParamList } from 'app/screens/app-screen'
+
+import { UserBadges } from '../user-badges/UserBadges2'
+
+const { getUser } = cacheUsersSelectors
+
+type ParamList = Pick<AppTabScreenParamList, 'Profile'>
+
+type UserLinkProps = Omit<TextLinkProps<ParamList>, 'to' | 'children'> & {
+  userId: ID
+}
+
+export const UserLink = (props: UserLinkProps) => {
+  const { userId, ...other } = props
+  const userName = useSelector((state) => getUser(state, { id: userId })?.name)
+
+  return (
+    <Flex direction='row' gap='xs' justifyContent='center' alignItems='center'>
+      <TextLink<ParamList>
+        to={{ screen: 'Profile', params: { id: userId } }}
+        numberOfLines={1}
+        shrink
```

Comment:
whoa what is this? does it set `flex-shrink`?

Reply:
ya! thinking it's a common enough issue that it may be good to have a shorthand. lmk what you think though. Prob at least worth a comment on the prop def

Reply:
would prefer it be named `flexShrink` personally to make it super clear. just my opinion.